packages/server/socketHelpers/sendEncodedMessage.ts

```diff
@@ -1,16 +1,57 @@
 import {HttpResponse, WebSocket} from 'uWebSockets.js'
+import handleDisconnect from '../socketHandlers/handleDisconnect'
 import sendSSEMessage from '../sse/sendSSEMessage'
+import ConnectionContext from './ConnectionContext'
 import isHttpResponse from './isHttpResponse'
 
 const ESTIMATED_MTU = 1400
-const sendEncodedMessage = (transport: WebSocket | HttpResponse, message: object | string) => {
-  if (transport.done) return
-  const str = JSON.stringify(message)
-  if (isHttpResponse(transport)) {
-    sendSSEMessage(transport as HttpResponse, str)
+const TIMEOUT_COEFFICIENT = 2.0
+const INITIAL_TIMEOUT = 2000 // 2 seconds
+const MAX_TIMEOUT = 10000 // 10 seconds
+const TIMEOUT_JITTER = 20 // 20 ms
+
+const sendAndPushToReliableQueue = (
```

Comment:
if this function took in the stringified payload like `JSON.stringify([object, mid])` then we wouldn't have to re-stringify it every time we resent it